src/github.com/getlantern/flashlight/client/reverseproxy.go

```diff
@@ -35,6 +35,9 @@ func (client *Client) newReverseProxy(bal *balancer.Balancer) *httputil.ReverseP
 		// field when upstream servers are trying to determine the client IP.
 		// We need to add also the X-Lantern-Device-Id field.
 		Director: func(req *http.Request) {
+			// Add back the Host header which was stripped by the ReverseProxy. This
+			// is needed for sites that do virtual hosting.
+			req.Header.Set("Host", req.Host)
```

Comment:
Yes, it's stripped by the http.Server, which is using the ReverseProxy as a Handler, so by the time it gets to the ReverseProxy's logic, the header is already gone.